summingbird-batch/src/main/scala/com/twitter/summingbird/batch/Batcher.scala

```
@@ -169,7 +169,13 @@ trait Batcher extends Serializable {
     dateToBatch(interval)(truncateUp)(truncateDown)
 
   def toInterval(b: BatchID): Interval[Timestamp] =
-    Intersection(InclusiveLower(earliestTimeOf(b)), ExclusiveUpper(earliestTimeOf(b.next)))
+    if (b.id < Long.MaxValue) {
```

Does this defense make sense? I think it makes code a little bit more noisy, while don't add value in most cases (if duration != 1ms).

---

So, the code is actually wrong if we don't do it. `b.next` will not be bigger than `b` if `b.id == Long.MaxValue`. And I don't want to have it be silently wrong.

If you prefer we can throw an error, I guess, which I don't like either.

Honestly, `earliestTimeOf(b)` can't be a total function, which is another correctness issues. It probably needs to return `Option[Timestamp]`, but that's another issue.

---

Can we use `batchIdSuccessible.next(b)` then?
If `earliestTimeOf` will return an `Option` it would be possible to `map` `next` with `earliestTimeOf` to make this function correct.

---

I'm not exactly sure what you are proposing here. We could write this code with a match:
```scala
batchIdSuccessible.next(b) match {
  case Some(bnext) => // can do left closed right open
  case None => // can't call next, so can only do inclusive
}
```
I'm not sure what `.map` has to do with it. I think you'd need a `.flatMap` actually since both `earliestTimeOf` might return Option and `next` should return `Option`.

Rather than do a big refactor of batcher and BatchID to deal with the partiality now, I just want to tackle fixing the things we can without changing the binary signature.

Note, the range bug I actually saw in a test (it would not terminate since previously it was an infinite loop).

---

Yes, I meant usage of `flatMap` in future, for now I'm ok with `match` and inclusive/exclusive intervals return.